### PR TITLE
[CHK-3296] feat: add mongo configuration for user-stats with mock

### PR DESCRIPTION
**Pull request by scaminati-bv**
#### List of Changes
- add mongo configuration
- add **user-stats** document into ecommerce db
- add service implementation to retrieve last used method data by userId
- add docker configuration with mock db

#### Motivation and Context
The goal of this PR is to add mongo configuration where users last payment method will be stored.
In addition, is added a service to retrieve the information of a specific user by id.

#### How Has This Been Tested?
Unit tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+MONGO_PASSWORD=password
+MONGO_SSL_ENABLED=false
+MONGO_MIN_POOL_SIZE=0
+MONGO_MAX_POOL_SIZE=20
+MONGO_MAX_IDLE_TIMEOUT_MS=60000
+MONGO_CONNECTION_TIMEOUT_MS=1000
+MONGO_SOCKET_TIMEOUT_MS=10000
+MONGO_SERVER_SELECTION_TIMEOUT_MS=2000
+MONGO_WAITING_QUEUE_MS=2000
+MONGO_HEARTBEAT_FREQUENCY_MS=5000
```
Comment (pietro-tota) [inline .env.example:17]: add those variables to README and helm values too (for helm values you can use the same values as done [here](https://github.com/pagopa/pagopa-ecommerce-transactions-service/blob/d50c375fdfe411bfc6988f4dea95f4811e02b353/helm/values-dev.yaml#L101-L112) :)

File: src/main/kotlin/it/pagopa/ecommerce/users/services/UserStatisticsService.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+@Service
+class UserStatisticsService(
+    @Autowired private val userStatisticsRepository: UserStatisticsRepository
+) {
+
+    fun findUserLastMethodById(userId: String): Mono<LastUsage> {
+        return userStatisticsRepository
+            .findById(userId)
+            .switchIfEmpty(Mono.error(Exception("User not found")))
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/users/services/UserStatisticsService.kt:17]:
```suggestion
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono

@Service
class UserStatisticsService(
    @Autowired private val userStatisticsRepository: UserStatisticsRepository
) {

    fun findUserLastMethodById(userId: String): Mono<LastUsage> {
        return userStatisticsRepository
            .findById(userId)
            .switchIfEmpty(Mono.error(UserNotFoundException(message = "", cause = null)))
```

use the custom exception here that is mapped to a 404 not found response in exception handler (see [here](https://github.com/pagopa/pagopa-ecommerce-user-stats-service/blob/1c0a86afc1572d73fa9fa539e3390d7cf68ad2a2/src/main/kotlin/it/pagopa/ecommerce/users/exceptions/handlers/ExceptionHandler.kt#L33-L44) and [here](https://github.com/pagopa/pagopa-ecommerce-user-stats-service/blob/main/src/main/kotlin/it/pagopa/ecommerce/users/exceptions/UserNotFoundException.kt))

**Review by pietro-tota — COMMENTED** [review]
🚀 well done, just some trivial issues :)

File: .env.example (hunk 1 above)
Comment (scaminati-bv) [reply .env.example:17]: done here 1ca48baac7ea9bfc539cd80c60a6371868f6d721

File: README.md [hunk 3]
```diff
@@ -62,11 +62,23 @@ to get a good default configuration.
...
+| MONGO_CONNECTION_TIMEOUT_MS       | Max time to wait for a connection to be opened. See docs *                                                                                                 | string |         |
+| MONGO_SOCKET_TIMEOUT_MS           | Max time to wait for a command send or receive before timing out. See docs *                                                                               | string |         |
+| MONGO_SERVER_SELECTION_TIMEOUT_MS | Max time to wait for a server to be selected while performing a communication with Mongo in milliseconds. See docs *                                       | string |         |
+| MONGO_WAITING_QUEUE_MS            | Max time a thread has to wait for a connection to be available in milliseconds. See docs *                                                                 | string |         |
+| MONGO_HEARTBEAT_FREQUENCY_MS      | Hearth beat frequency in milliseconds. This is an hello command that is sent periodically on each active connection to perform an health check. See docs * | string |         |
```
Comment (pietro-tota) [inline README.md:81]:
```suggestion
| MONGO_HEARTBEAT_FREQUENCY_MS      | Hearth beat frequency in milliseconds. This is an hello command that is sent periodically on each active connection to perform an health check. See docs * | string |         |

(*): for Mongo connection string options
see [docs](https://www.mongodb.com/docs/drivers/java/sync/v4.3/fundamentals/connection/connection-options/#connection-options)
```

File: helm/values-dev.yaml [hunk 4]
```diff
@@ -87,6 +87,18 @@ microservice-chart:
     ROOT_LOGGING_LEVEL: "INFO"
     APP_LOGGING_LEVEL: "INFO"
     WEB_LOGGING_LEVEL: "OFF"
+    MONGO_HOST: pagopa-d-weu-ecommerce-cosmos-account.mongo.cosmos.azure.com
```
Comment (pietro-tota) [inline helm/values-dev.yaml:90]: missing MONGO_PASSWORD env property (to be add as a secret, see [here](https://github.com/pagopa/pagopa-ecommerce-transactions-service/blob/d50c375fdfe411bfc6988f4dea95f4811e02b353/helm/values-dev.yaml#L186) for an example).

Please add it to helm values for all envs

**Review by pietro-tota — APPROVED** [review]

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**